src/main/java/de/tum/in/www1/artemis/repository/StudentQuestionAnswerRepository.java

```diff
@@ -16,7 +16,7 @@
 public interface StudentQuestionAnswerRepository extends JpaRepository<StudentQuestionAnswer, Long> {
 
     @NotNull
-    default StudentQuestionAnswer findByIdElseThrow(long studentQuestionId) {
+    default StudentQuestionAnswer findByAnswerIdElseThrow(long studentQuestionId) {
         return findById(studentQuestionId).orElseThrow(() -> new EntityNotFoundException("Student Question Answer", studentQuestionId));
```

Comment:
```suggestion
    default StudentQuestionAnswer findByAnswerIdElseThrow(long studentQuestionAnswerId) {
        return findById(studentQuestionAnswerId).orElseThrow(() -> new EntityNotFoundException("Student Question Answer", studentQuestionAnswerId));
```

Naming the function `findByAnswerId` with a parameter `questionId` is a bit confusing.

Reply:
I would suggest that we stick to `findByIdElseThrow` here, because it is obvious from the repository what we would like to fetch from the database. To make it easier to understand, please user `answerId` as parameter name